packages/syft/src/syft/core/node/common/action/smpc_action_message.py

```diff
@@ -64,13 +70,35 @@ def filter_actions_after_rank(
         """
         res_actions = []
         for action in actions:
-            if action.ranks_to_run_action is None:
+
+            should_not_run = (
+                isinstance(action.ranks_not_to_run_action, list)
+                and rank in action.ranks_not_to_run_action
+            )
+
+            should_run = False
+            if (
+                action.ranks_to_run_action == []
```

Comment:
Should we just check the length in case we have a different type of list or iterator which doesnt equal empty list []?

Reply:
Also, is there a reason the default value for ranks_to and ranks_not_to are None and not empty lists? Is it correct that this should run if ranks is an empty list but not if it's None?